src/components/custom-nav-bar.js

```diff
@@ -12,6 +12,28 @@ class CustomNavbar extends React.Component {
     super(props);
   }
 
+  navlinks(props) {
+    let token = props.token;
+    if(token === null) {
+      return (
+          <>
+            <Nav.Link href="#about">About</Nav.Link>
```

Comment:
I think one of the react's advantages is that it makes a SPA (Single Page Application). If you click Nav.Link, I found it refreshes the page. Refreshing makes the application loses `Context` and other things in the memory. So I suggest to use `<Nav.Link as={Link} to="/">`. I found it from https://stackoverflow.com/questions/59530374/react-boostrap-and-react-router-causing-full-page-reload.